src/mono/wasm/host/DevServer/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,138 @@
+            subBuilder =>
+            {
+                subBuilder.Use(async (context, next) =>
+                {
+                    context.Response.Headers.Append("DotNet-Environment", webHostEnvironment.EnvironmentName);
+
+                    // DOTNET_MODIFIABLE_ASSEMBLIES is used by the runtime to initialize hot-reload specific environment variables and is configured
+                    // by the launching process (dotnet-watch / Visual Studio).
+                    // Always add the header if the environment variable is set, regardless of the kind of environment.
+                    if (s_dotnetModifiableAssemblies != null)
+                    {
+                        context.Response.Headers.Append("DOTNET-MODIFIABLE-ASSEMBLIES", s_dotnetModifiableAssemblies);
+                    }
+
+                    // See https://github.com/dotnet/aspnetcore/issues/37357#issuecomment-941237000
+                    // Translate the _ASPNETCORE_BROWSER_TOOLS environment configured by the browser tools agent in to a HTTP response header.
+                    if (s_aspnetcoreBrowserTools != null)
+                    {
+                        context.Response.Headers.Append("ASPNETCORE-BROWSER-TOOLS", s_aspnetcoreBrowserTools);
+                    }
+
+                    await next(context);
+                });
+
+                subBuilder.UseMiddleware<ContentEncodingNegotiator>();
+
+                subBuilder.UseStaticFiles(options);
+            }
+        );
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures the application to serve Blazor WebAssembly framework files from the root path "/".
+    /// </summary>
+    /// <param name="applicationBuilder">The <see cref="IApplicationBuilder"/>.</param>
+    /// <returns>The <see cref="IApplicationBuilder"/></returns>
+    public static IApplicationBuilder UseBlazorFrameworkFiles(this IApplicationBuilder applicationBuilder) =>
+        UseBlazorFrameworkFiles(applicationBuilder, default);
+
+    private static StaticFileOptions CreateStaticFilesOptions(IFileProvider webRootFileProvider)
+    {
+        var options = new StaticFileOptions();
+        options.FileProvider = webRootFileProvider;
+        var contentTypeProvider = new FileExtensionContentTypeProvider();
+        AddMapping(contentTypeProvider, ".dll", MediaTypeNames.Application.Octet);
+        AddMapping(contentTypeProvider, ".webcil", MediaTypeNames.Application.Octet);
+        // We unconditionally map pdbs as there will be no pdbs in the output folder for
+        // release builds unless BlazorEnableDebugging is explicitly set to true.
+        AddMapping(contentTypeProvider, ".pdb", MediaTypeNames.Application.Octet);
+        AddMapping(contentTypeProvider, ".br", MediaTypeNames.Application.Octet);
+        AddMapping(contentTypeProvider, ".dat", MediaTypeNames.Application.Octet);
+        AddMapping(contentTypeProvider, ".blat", MediaTypeNames.Application.Octet);
```

Comment:
We could remove `.blat`

Reply:
Theoretically. My longer term goal is to use WasmAppHost even for Blazor projects and than we would nee `.blat` from hosting .NET 7 Blazor.  I'm not sure yet if that is doable path. Leaving the extension there probably doesn't do any harm.